jerry-core/jrt/jrt-bit-fields.h

```diff
@@ -16,7 +17,28 @@
 #ifndef JERRY_BIT_FIELDS_H
 #define JERRY_BIT_FIELDS_H
 
-extern uint64_t __attr_const___ jrt_extract_bit_field (uint64_t, size_t, size_t);
-extern uint64_t __attr_const___ jrt_set_bit_field_value (uint64_t, uint64_t, size_t, size_t);
+/**
+ * Extract a bit-field.
+ *
+ * @param container container to extract bit-field from
+ * @param lsb least significant bit of the value to be extracted
+ * @param width width of the bit-field to be extracted
+ * @return bit-field's value
+ */
+#define JRT_EXTRACT_BIT_FIELD(container, lsb, width) \
+  (((container) >> lsb) & ((((__typeof__ (container)) 1) << (width)) - 1))
```

Comment:
Now, that you mention... Interestingly, the build system does not complain about it. I see two ways of dealing with this:
* leave it in the currently proposed form, or
* let the macros have one more argument requiring the type of the container explicitly.

This first approach is a bit less standard-conforming (but seems to be supported by all compiilers out there, at least by clang for sure), while the second one needs more maintenance if the type of the first argument changes (anywhere where the macro is used).

Opinions?